app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,31 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: [:create, :destroy]
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build micropost_params
+    if micropost.save
+      flash[:success] = t "micropost_created"
+      redirect_to root_url
+    else
+      @feed_items = current_user.feed.paginate page: params[:page]
+      render "static_pages/home"
+    end
+  end
+
+  def destroy
+    micropost.destroy
+    flash[:success] = t "micropost_deleted"
+    redirect_to request.referer || root_url
+  end
+
+  private
+  def micropost_params
+    params.require(:micropost).permit :content, :picture
+  end
+
+  def correct_user
+    @micropost = current_user.microposts.find_by id: params[:id]
+    redirect_to root_url if micropost.blank?
```

Comment:
nil? chỉ dùng cho object, trả về true nếu object là nil
empty? chỉ dùng cho array, hash, string, trả về true khi độ dài các cái trên = 0
blank? trả về true nếu object là nil hoặc false hoặc là empty tring, hash, array hoặc là 1 string chỉ toàn khoảng trắng